internal/terraform/resource_test.go

```diff
@@ -19,16 +19,43 @@ import (
 	"github.com/terraform-docs/terraform-docs/internal/types"
 )
 
-func TestResourceFullType(t *testing.T) {
+func TestResourceSpec(t *testing.T) {
 	assert := assert.New(t)
 	resource := Resource{
 		Type:           "private_key",
+		Name:           "baz",
 		ProviderName:   "tls",
 		ProviderSource: "hashicorp/tls",
 		Mode:           "managed",
 		Version:        types.String("latest"),
 	}
-	assert.Equal("tls_private_key", resource.FullType())
+	assert.Equal("tls_private_key.baz", resource.Spec())
+}
+
+func TestGetModeAsManagedMode(t *testing.T) {
```

Comment:
Please combine `TestGetModeAsManagedMode` and `TestGetModeAsDataMode` into one function called `TestResourceMode`.